config.go

```diff
@@ -0,0 +1,85 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"go.uber.org/multierr"
+)
+
+// Config has the generic parameters needed for a gRPC client
+type Config struct {
+	// url to gRPC server
+	URL string `json:"url" validate:"required"`
+	// the bandwidth limit in bytes/second, use "0" to disable rate limiting.
+	RateLimit int `json:"rateLimit" default:"0" validate:"gt=-1"`
+	// mTLS configurations.
+	MTLS MTLSConfig `json:"mtls"`
+}
+
+type MTLSConfig struct {
+	// the client certificate path.
+	ClientCertPath string `json:"client.certPath"`
+	// the client private key path.
+	ClientKeyPath string `json:"client.keyPath"`
+	// the root CA certificate path.
+	CACertPath string `json:"CA.certPath"`
```

Comment:
I know CA is an abbreviation, I'd still use lowercase `ca` for the config field, we also use `mtls` and `url` for instance.

Reply:
I was thinking about this and convinced myself that `ca` is better and refactored it, then convinced myself that `CA` is better and reverted everything 🫠
BUT the url arguments convinced me again :D